src/main/java/com/google/devtools/build/lib/windows/WindowsSubprocessFactory.java

```diff
@@ -45,6 +49,14 @@ public Subprocess create(SubprocessBuilder builder) throws IOException {
     String argv0 = processArgv0(argv.get(0));
     String argvRest =
         argv.size() > 1 ? WindowsProcesses.quoteCommandLine(argv.subList(1, argv.size())) : "";
+
+    if (argvRest.length() + 2 > MAX_ARGUMENT_LENGTH) { // +2 for unicode null terminator
```

Comment:
You need to account for argv0's length too.

Reply:
Also, I think +2 is too much, +1 is enough, because String.length() returns the length in character count, not in bytes.

Reply:
Okay; [the docs](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682425(v=vs.85).aspx) only mention a restriction on the `lpCommandLine` but I'll double check this.

I agree with you, I think `+2` is wrong. I'll make the tests and confirm that the behaviour is correct. Thanks for the pointer!

Reply:
@j3parker : Are you still working on this PR?

Reply:
@laszlocsomor oh hey, I forgot about this. I paused because my Bazel build on windows wasn't working.

Yes I'll finish it up, hopefully tomorrow.

Reply:
Great, thanks!